tests/unit/field/test_field2d.cxx

```diff
@@ -447,6 +448,49 @@ TEST_F(Field2DTest, CheckData) {
   EXPECT_THROW(checkData(field), BoutException);
 }
 
+TEST_F(Field2DTest, InvalidateGuards) {
+  Field2D field;
+  field.allocate(); // Calls invalidateGuards
+  field = 1.0;      // Sets everywhere including boundaries
+
+  const int nmesh = nx * ny;
+
+  int sum = 0;
+  for (const auto &i : field.region(RGN_ALL)) {
+    field[i] = 0.0; // Reset field value
+    sum++;
+  }
+  EXPECT_EQ(sum, nmesh); // Field operator= hasn't been broken by invalidateGuards
+
+  // Count the number of non-boundary points
+  sum = 0;
+  for (const auto &i : field.region(RGN_NOBNDRY)) {
+    field[i] = 0.0; // Reset field value
+    sum++;
+  }
+  const int nbndry = nmesh - sum;
+
+#if CHECK > 2
```

Comment:
What check level is the test suite run at?

Reply:
It depends on the user's configure choices (e.g. the travis builds includes CHECK=0, CHECK=3 and the default of CHECK=2 [I think]).

Reply:
ok, just checking that this is tested on Travis. Thanks!